plugin/transfer/transfer.go

```diff
@@ -43,6 +51,12 @@ type Transferer interface {
 	// If the serial is less (older) than the current serial for the zone, perform an AXFR fallback
 	// by proceeding as if an AXFR was requested (as above).
 	Transfer(zone string, serial uint32) (<-chan []dns.RR, error)
+
+	// Notify returns a data channel to which the implementing plugin should send a message for each zone
+	// that has updated, and a stop channel.  The transfer plugin listens for messages and sends notifications to
+	// the destinations in "to".  A plugin that does not want to support notifications can return nil
+	// channels.
+	Notify() (data <-chan []string, stop <-chan struct{})
```

Comment:
I think just exporting a function as done in https://github.com/coredns/coredns/pull/3972 is simpler, and side steps the whole ping. During setup you just grab the transfer plugin and then you call Notify() when ever you need it.